### DISCUSSION TITLE
Internationalization(i18n) discussion

### DISCUSSION THREAD

**Discussion by ycyy** [Comfy-Org/ComfyUI_frontend#417] [General]
The current internationalization scheme is written in a single file, I think it should be split into multiple files and then introduced individually via import. At the same time, we should play the role of the community, and utilize the translation platform like [crowdin](https://crowdin.com/) so that we can translate into the local language together.

**Comment by huchenlei** [comment]
Thanks for the suggestion! The i18n.ts setup is pretty temporary. I would definitely like to hear more about how to get the community involved with this i18n project. There is a ComfyUI translation project already, so probably we can leverage that as well. https://github.com/AIGODLIKE/AIGODLIKE-ComfyUI-Translation

I created a public project on crowdin here: https://crowdin.com/project/comfyuifrontend. Please feel free to suggest what we need to do next!

**Reply by ycyy** [reply]
[AIGODLIKE-ComfyUI-Translation](https://github.com/AIGODLIKE/AIGODLIKE-ComfyUI-Translation) is a good project, but it's built on an old front-end framework. It should be used to build a new internationalization, drawing on its experience as the new project is gradually being refined.I'm looking forward to it.

**Comment by wfjsw** [comment]
Depends on whether you want all localizations to reside in a single extension or have each extensions provide their own localization (which makes more sense) I would recommend using https://github.com/i18next/i18next-vue accompanying with a proper backend implementation (e.g. https://github.com/i18next/i18next-http-backend), where each extension has their own namespace.

In any case there should be proper stub within litegraph. Hacking all the way into labels breaks the original functionality.
